config/crds/apis.kcp.dev_apibindings.yaml

```diff
@@ -69,6 +69,14 @@ spec:
                   required:
                   - resource
                   type: object
+                  x-kubernetes-validations:
+                  - message: .resource must be one of configmaps, namespaces, secrets,
+                      serviceaccounts if .group is empty.
+                    rule: (has(self.group) && self.group != '') || self.resource in
+                      ['configmaps', 'namespaces', 'secrets', 'serviceaccounts']
+                  - message: .identityHash must be set if .group is not empty.
+                    rule: '!(has(self.group) && self.group != '''') || (has(self.identityHash)
+                      && self.identityHash != '''')'
```

Comment:
much better

Reply:
I honestly have never done logic like this or at least forgot that it was possible. Thanks for teaching!

Reply:
https://en.wikipedia.org/wiki/Conjunctive_normal_form